### PR TITLE
py-random2: update to 1.0.2, drop unsupported pythons

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: python/py-random2/Portfile [hunk 1]
```diff
@@ -14,11 +14,10 @@ description         Python 3 compatible Python 2 random Module.
...
 
-checksums           rmd160  e0c8cac5bf246d6ee776c1ea7ac47baaf7a3e8ab \
-                    sha256  34ad30aac341039872401595df9ab2c9dc36d0b7c077db1cea9ade430ed1c007 \
-                    size    21088
+checksums           rmd160  af0574efdade2c422987e5091d3f1c6cc69dab0f \
+                    sha256  3754fcef48267567cd5705fa7da6bbc3809cb3f808740313e6705acc3c057e77 \
+                    size    18095
 
-python.versions     38 39 310 311 312
+python.versions     310 311 312
```
[the remark this answers is not in the document]
Comment (reneeotten) [reply python/py-random2/Portfile:23]: @barracuda156 please address this so we can merge the PR

[the remark this answers is not in the document]
Comment (reneeotten) [reply python/py-random2/Portfile:23]: ping

[the remark this answers is not in the document]
Comment (barracuda156) [reply python/py-random2/Portfile:23]: Ok, lumme restore those.

[the remark this answers is not in the document]
Comment (barracuda156) [reply python/py-random2/Portfile:23]: @reneeotten Ah, by unsupported I meant unsupported by upstream now: https://pypi.org/project/random2

> Change support to Python 3.10, 3.11, 3.12, and 3.13 only.

Add a fallback version for older Pythons?

[the remark this answers is not in the document]
Comment (reneeotten) [reply python/py-random2/Portfile:23]: or you check the port that needs them for their test phase and see if they are really needed and/or if you can just disable the test for the older Python versions. I would rather not add fallback versions....

[the remark this answers is not in the document]
Comment (reneeotten) [reply python/py-random2/Portfile:23]: I update the dependent ports, and the test dependency is no longer needed. So it's all good now.

[the remark this answers is not in the document]
Comment (barracuda156) [reply python/py-random2/Portfile:23]: Ok great, thanks

**Pull request merged**